Production/src/Selectors.cc

```diff
@@ -257,7 +257,11 @@ TauJetSelector::Result TagAndProbe::Select(const edm::Event& event, const std::d
         	tagObject = nullptr;   
     	}
         const TauJet* selected = nullptr;
-        for(const TauJet& tauJet : tauJets) { //Look for highest pT jet
+        for(const TauJet& tauJet : tauJets) { //Look for highest pT jet and for gen tag 
+            if(tauJet.genLepton && tagObject != nullptr && reco::deltaR(tauJet.genLepton->visibleP4(), tagObject->p4) < 0.1) {
+	    	    tagObject->genkind = tauJet.genLepton->kind();
+		    tagObject->genp4 = tauJet.genLepton->visibleP4();
+	    }
```

Comment:
Maybe I misunderstood the logic, but I think the following case might not have correct flow:
1) One has tag e/mu
2) There is no jet that  satisfy this criteria: 
```c++
            if(!(std::abs(jet.eta()) < 2.8) || ((ref_muon || ref_electron) && !(reco::deltaR(jet.p4(), tagObject->p4) > 0.4)))
                    continue;
```
3) There are still gen level objects that will be saved
4) Because tagObject is not nullptr some information about it might also be saved. Do we really need to save it?